src/LobbyServer.hs

```diff
@@ -119,26 +119,29 @@ getGamesList remoteGames = do
   return $ map fst gameList
 
 -- |Lets a player join a 'LobbyGame'
-playerJoinGame :: Server ConcurrentClientList -> Server GamesList -> String -> Server ()
+playerJoinGame :: Server ConcurrentClientList -> Server GamesList -> String -> Server Bool
 playerJoinGame remoteClientList remoteGameList gameID = do
   clientList <- remoteClientList >>= liftIO . CC.readMVar
   gameList <- remoteGameList
   sid <- getSessionID
-  case lookupClientEntry sid clientList of
-    Just player -> liftIO $ CC.modifyMVar_ gameList $
-      \gList -> return $ addPlayerToGame player gameID gList
-
-    _ -> return ()
-
-  maybeGame <- findGameWithSid gameList
+  maybeGame <- liftIO $ findGameWithID gameID gameList
   case maybeGame of
-    Just (_,gameData) -> liftIO $ messageClients PlayerJoinedGame (players gameData)
-    nothing           -> return ()
+    Nothing           -> return False
+    Just (_,gameData) ->
+      if maxAmountOfPlayers gameData > length (players gameData) then do
+          case lookupClientEntry sid clientList of
+            Nothing     -> error "playerJoinGame: Client not registered"
```

Comment:
Actually wanted to crash the server :sunglasses: (as it is an unallowed state)

Seriously though, the error isn't relevant to the client so I see no point in passing it there. We do however need some way of logging errors of this sort. I do however agree that it should simply be removed. If the Nothing case occurs, it is probably because of the client initiating the connection and then exiting the lobby, at which point we can just swallow the error. Will remove. :bug: :beetle: